src/test/kotlin/com/exactpro/th2/processor/cache/collector/message/TestCacheMessage.kt

```diff
@@ -76,34 +80,45 @@ class TestCacheMessage {
         .setMetadata(rawMessageMetadata)
         .build()
 
+    private fun GrpcParsedMessage.getBody(): Map<String, Any> {
+        val jsonString = CustomProtoJsonFormatter().print(this)
+        return ObjectMapper().readValue(jsonString, Map::class.java) as Map<String, Any>
+    }
+
     private fun compare(cacheParsedMessage: ParsedMessage) {
-        assert(cacheParsedMessage.book == messageId.bookName)
-        assert(cacheParsedMessage.group == grpcMessage.sessionGroup)
-        assert(cacheParsedMessage.sessionAlias == grpcMessage.sessionAlias)
-        assert(cacheParsedMessage.direction == grpcMessage.direction.toString())
-        assert(cacheParsedMessage.sequence == grpcMessage.sequence)
-        assert(cacheParsedMessage.subsequence == grpcMessage.subsequence)
-        assert(cacheParsedMessage.metadata.messageType == grpcMessage.metadata.messageType)
-        assert(cacheParsedMessage.metadata.protocol == grpcMessage.metadata.protocol)
+        assertEquals(cacheParsedMessage.book, messageId.bookName)
```

Comment:
Raw & parsed messages must be parameters of compare method.
Their construction must happen inside test cases